### PR TITLE
Peerdas: Several updates

### PR DESCRIPTION
**Pull request by nalepae**
**Please read commit by commit**

- Add more logs regarding disconnection
- Add a new hidden flag `--data-columns-reject-slot-multiple`
- Do some refactor

### CODE REVIEW

File: beacon-chain/p2p/handshake.go [hunk 1]
```diff
@@ -22,7 +22,57 @@ const (
...
+	// Get the remaining active peers.
+	activePeerCount := len(s.peers.Active())
+	log.
+		WithError(badPeerErr).
+		WithFields(logrus.Fields{
+			"multiaddr":            remotePeerMultiAddr,
+			"direction":            direction,
+			"remainingActivePeers": activePeerCount,
+		}).
+		Debug("Initiate peer disconnection")
```
Comment (nisdas) [inline beacon-chain/p2p/handshake.go:73]: This will be noisy on mainnet, do we really need to log this ?

File: beacon-chain/p2p/peers/status.go [hunk 2]
```diff
@@ -800,134 +811,87 @@ func (p *Status) BestNonFinalized(minPeers int, ourHeadEpoch primitives.Epoch) (
 // bad response count. In the future scoring will be used
 // to determine the most suitable peers to take out.
 func (p *Status) PeersToPrune() []peer.ID {
```
Comment (nisdas) [inline beacon-chain/p2p/peers/status.go:813]: This is a sensitive function, can you refactor this in a separate PR ?

File: beacon-chain/p2p/handshake.go (hunk 1 above)
Comment (nalepae) [reply beacon-chain/p2p/handshake.go:73]: At least for devnet, it's important to know why a peer is disconnected.
Is it from our initiative? Is it from the peer initiative?
If it is from our initiative, what was the reason? Bad gossip score? Too many bad response? Other?

If it is too noisy on mainnet, or even on testnet, I can only print this log only if we are not on mainnet/testnet.

File: beacon-chain/p2p/peers/status.go (hunk 2 above)
Comment (nalepae) [reply beacon-chain/p2p/peers/status.go]: Done in https://github.com/prysmaticlabs/prysm/pull/14460.

File: beacon-chain/p2p/handshake.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/p2p/handshake.go:73]: We could leave this in the `peerDAS` branch if it helps with debugging but not bring it down to `develop`

**Review by nisdas — APPROVED** [review]

**Pull request merged**